### PR TITLE
Publish to npm on PR merge into `main`

### PR DESCRIPTION
**Pull request by AndrewGable**
### Details

Publishes to NPM registry on PR merge into `main`. After merge it will:
1. Create a new branch with a UUID
2. Update the version (this also tags the commit automatically)
3. Build the web code
4. Publish to NPM
5. Merge the branch created in step 1

### Related Issues

https://github.com/Expensify/Expensify/issues/211122

### Automated Tests

1. Merge this PR once it's off hold
2. It should create version 1.0.2

### CODE REVIEW

**Comment by AndrewGable** [comment]
Alright 🤞 this works, it usually doesn't on the first run, but should be good to merge and try it out now.

**Review by tgolen — DISMISSED** [review]

File: package.json
Comment (parasharrajat) [inline package.json:1]: We should add keywords, repository url. E.g.

```js
{
  "name": "my_package",
  "description": "",
  "version": "1.0.0",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/monatheoctocat/my_package.git"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/monatheoctocat/my_package/issues"
  },
  "homepage": "https://github.com/monatheoctocat/my_package"
```

File: .github/workflows/publish.yml
Comment (tgolen) [reply .github/workflows/publish.yml:42]: Oh, now the webpack configs are all exported from the same file, so it only needs one build command. Let's just rename it from `build:web` to `build` since it's no longer web specific.

Comment (AndrewGable) [reply .github/workflows/publish.yml]: 👍

**Comment by danieldoglas** [comment]
This LGTM, but I'll let you merge so you can see the magic happening :) @AndrewGable

**Review by danieldoglas — APPROVED** [review]
LGTM

**Comment by AndrewGable** [comment]
Tagging in @roryabraham for one final review before testing.

**Comment by AndrewGable** [comment]
Ok off hold @roryabraham

File: .github/workflows/publish.yml
Comment (roryabraham) [inline .github/workflows/publish.yml:83]: Again, let's just use the gh cli (note that the approver and PR creator must be different tokens)
```suggestion
            - name: Auto-approve pull request
              run: gh pr review --approve
              env:
                GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/publish.yml
Comment (roryabraham) [inline .github/workflows/publish.yml:90]:
```suggestion
            - name: Auto-merge pull request
              run: gh pr merge --merge --delete-branch
              env:
                GITHUB_TOKEN: ${{ secrets.BOTIFY_TOKEN }}
```

File: .github/workflows/publish.yml
Comment (roryabraham) [reply .github/workflows/publish.yml:71]: Also note that if we use the gh cli to merge the pull request, then we don't need the `automerge` label

**Review by roryabraham — CHANGES_REQUESTED** [review]

**Comment by parasharrajat** [comment]
Should we commit the GPG key into the repo?

**Comment by AndrewGable** [comment]
@parasharrajat - Encrypted, but yes

**Comment by AndrewGable** [comment]
Updated to use GH cli for everything but I am a bit worried won't know what pull request to approve or merge, so looking into that

File: .github/workflows/publish.yml
Comment (roryabraham) [inline .github/workflows/publish.yml:64]: NAB this step is missing a `name`

**Review by roryabraham — APPROVED** [review]

**Comment by roryabraham** [comment]
Never works first try 😭 https://github.com/Expensify/react-native-onyx/runs/6617714651?check_suite_focus=true

**Comment by marcaaron** [comment]
Hell yes !! I have had [this issue](https://github.com/Expensify/App/issues/4680) open forever.

**Comment by marcaaron** [comment]
Thanks for this change!!!

Also, this is happening which is the only reason why I noticed -> 

https://github.com/Expensify/react-native-onyx/runs/6618815651?check_suite_focus=true

**Comment by roryabraham** [comment]
Thanks! We are fixing it, slowly but surely 😆

**Pull request merged**
